File: src/main/java/tomeko/chatblock/ChatBlock.java
package tomeko.chatblock;

import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import tomeko.chatblock.chat.*;
import tomeko.chatblock.config.*;

@Mod(modid = ChatBlock.MODID, name = ChatBlock.NAME, version = ChatBlock.VERSION)
public class ChatBlock {
	public static final String MODID = "@ID@";
	public static final String NAME = "@NAME@";
	public static final String VERSION = "@VER@";

	@Mod.Instance(MODID)
	public static ChatBlock INSTANCE;
	public static ChatBlockConfig config;

	@Mod.EventHandler
	public void onInit(FMLInitializationEvent event) {
		config = new ChatBlockConfig();

		Chat.register();
	}
}

File: src/main/kotlin/tomeko/chatblock/ChatBlock.kt
package tomeko.chatblock

//? if forge {
/*import cc.polyfrost.oneconfig.events.EventManager;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
*///?} elif ornithe {
//import net.ornithemc.osl.entrypoints.api.ModInitializer
//?} else {
import net.fabricmc.api.ClientModInitializer
//?}
import tomeko.chatblock.chat.*
import tomeko.chatblock.commands.*
import tomeko.chatblock.config.*
import tomeko.chatblock.utils.*

//? if forge {
/*@Mod(
    modid = Constants.MOD_ID,
    name = Constants.MOD_NAME,
    version = Constants.MOD_VERSION,
    modLanguageAdapter = "cc.polyfrost.oneconfig.utils.KotlinLanguageAdapter"
)
*///?}
class ChatBlock
//? if ornithe {
    //: ModInitializer
//?} elif fabric {
    : ClientModInitializer
//?}
{
    //? if forge {
    //@Mod.EventHandler
    //?} else {
    override
    //?}
    fun
            //? if ornithe {
            //init(
        //?} else {
            onInitializeClient(
        //?}
        //? if forge {
        //event: FMLInitializationEvent
        //?}
    ) {
        //? if forge {
        //EventManager.INSTANCE.register(this)
        //?}

        BlockReceivingMessages.register()
        BlockSendingWords.register()

        ChatBlockCommand.register()

        ChatBlockConfig.register()

        Debug.forceLog("Initialized!")
    }
}